### PR TITLE
Toradex updates for sumo

### PR DESCRIPTION
**Pull request by drewmoseley**

### CODE REVIEW

**Comment by drewmoseley** [comment]
@mirzak ping

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc [hunk 1]
```diff
@@ -33,3 +33,5 @@ SRC_URI_append_colibri-imx7 = " \
 BOOTENV_SIZE_colibri-imx7-emmc ?= "0x2000"
 
 SRC_URI_append_colibri-imx7-emmc = " file://0001-colibri-imx7-emmc-mender-integration-patch.patch"
+
+MENDER_UBOOT_PRE_SETUP_COMMANDS_apalis-imx6 = "setenv loadaddr \${kernel_addr_r}; run setup; setenv bootargs \${defargs} \${emmcargs} \${setupargs} \${vidargs}; "
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc]: Missing newline at the end of this file

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-toradex-fsl-fw-utils_%.bbappend [hunk 2]
```diff
@@ -2,3 +2,5 @@
 # u-boot-mender-common instead of u-boot-fw-utils-mender
 require recipes-bsp/u-boot/u-boot-mender-common.inc
 require u-boot-mender-toradex-nxp.inc
+
+SRC_URI_append_apalis-imx6 = " file://0001-apalis-imx6-mender-integration-patch.patch "
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-toradex-fsl-fw-utils_%.bbappend]: Missing newline at the end of this file

File: meta-mender-toradex-nxp/templates/local.conf.append [hunk 3]
```diff
@@ -70,3 +72,5 @@ MACHINE_ESSENTIAL_EXTRA_RDEPENDS = "kernel-image kernel-devicetree"
 # pick u-boot-toradex-fsl-fw-utils as provider of fw_{printenv,setenv}
 PREFERRED_PROVIDER_u-boot-fw-utils ?= "u-boot-toradex-fsl-fw-utils"
 PREFERRED_RPROVIDER_u-boot-fw-utils ?= "u-boot-toradex-fsl-fw-utils"
+PREFERRED_VERSION_imx-gpu-viv = "6.2.4.p1.2-aarch32"
+PREFERRED_VERSION_kernel-module-imx-gpu-viv = "6.2.4.p1.2"
```
Comment (mirzak) [inline meta-mender-toradex-nxp/templates/local.conf.append]: How is this related to the Mender integration?

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ #if defined(CONFIG_ENV_IS_IN_NAND)
+ #define CONFIG_ENV_SECT_SIZE		(128 * 1024)
+-/* configured by mender integration */
+-/* #define CONFIG_ENV_OFFSET		(28 * CONFIG_ENV_SECT_SIZE) */
+-#define CONFIG_ENV_SIZE			CONFIG_ENV_SECT_SIZE
+-#define CONFIG_ENV_RANGE    (2 * CONFIG_ENV_SIZE)
++/* modified for mender integration */
++#define CONFIG_ENV_OFFSET		0x380000
++#define CONFIG_ENV_SIZE                 0x18000
++#define CONFIG_ENV_RANGE                0x40000
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: Why can not let these be defined by the Yocto recipes as it was previously?

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-tools-env-Add-support-for-direct-read-write-UBI-volu.patch [hunk 5]
```diff
@@ -0,0 +1,374 @@
+From b95b617a0a112d1f1a9b22c90bfd90609e21e04e Mon Sep 17 00:00:00 2001
+From: "S. Lockwood-Childs" <sjl@vctlabs.com>
+Date: Tue, 14 Nov 2017 23:01:26 -0800
+Subject: [PATCH] tools: env: Add support for direct read/write UBI volumes
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-tools-env-Add-support-for-direct-read-write-UBI-volu.patch]: This is added, but I can not see that is being used by the imx7 integration?

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch [hunk 6]
```diff
@@ -0,0 +1,34 @@
+From b39a6a374cd6ebb9c9bc1ca17fc7f59364500f5a Mon Sep 17 00:00:00 2001
+From: Drew Moseley <drew.moseley@northern.tech>
+Date: Fri, 6 Dec 2019 15:07:25 +0000
+Subject: [PATCH] colibri-imx7: Adapt u-boot environment settings for Mender on
+ sumo.
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: I have been running imx7 on sumo for a while now. Not sure why these changes are needed?

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc [hunk 7]
```diff
@@ -20,13 +20,15 @@ SRC_URI_append_vf = " \
 
 SRC_URI_append_mx7 = " file://0001-Set-default-display-resolution-to-800x480-WVGA.patch"
 
-# 128kB
-BOOTENV_SIZE_colibri-imx7 ?= "0x20000"
+# 128kB - a bit to ensure 2 copies fit in 256KB
+BOOTENV_SIZE_colibri-imx7 ?= "0x18000"
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc]: Why?

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc (hunk 1 above)
Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-mender-toradex-nxp.inc]: Fixed

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-toradex-fsl-fw-utils_%.bbappend (hunk 2 above)
Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/u-boot-toradex-fsl-fw-utils_%.bbappend]: Fixed

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch (hunk 6 above)
Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: Which branch of the Toradex layers are you using?

Comment (mirzak) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: I have been using the versions that are in the current manifest,

https://github.com/mendersoftware/meta-mender-community/blob/sumo/meta-mender-toradex-nxp/scripts/manifest-toradex.xml

Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: OK.  I looked in the repos themselves and didn't see a sumo branch so I was using rocko-next (I think).  I'll take another pass through with that manifest.

Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: Oh.  I just realized that was the meta-mender-community manifest and not one from Toradex. How did you identify those SHAs?

Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: Are you sure have used sumo and not rocko? I tried to do a clean checkout and build and get the following:

``
$ source setup-environment toradex-nxp
Sorry, it does not seem that *toradex-nxp* is a valid target

Supported targets are:
atmel
beaglebone
intel
nxp
qemu
raspberrypi
renesas
rockchip
sunxi
tegra
up
variscite
``

Comment (mirzak) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: > Are you sure have used sumo and not rocko?

Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: OK.  I still have to change the bootenv size to get it to build.  Are you seeing that?

Comment (mirzak) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: Nope, build passes just fine. What specific error are you getting?

Comment (drewmoseley) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: I think I know why we are each confused here. I'm building colibri-imx7 which has failures and you are building colibri-imx7-emmc which does not.

Comment (mirzak) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/files/colibri-imx7/0001-colibri-imx7-Adapt-u-boot-environment-settings-for-M.patch]: > I think I know why we are each confused here. I'm building colibri-imx7 which has failures and you are building colibri-imx7-emmc which does not.

True

**Comment by drewmoseley** [comment]
@mirzak I redid this against the same commits you were using.  Note as discussed previously that these changes are for colibri-imx7 and not colibri-imx7-emmc.
